### DISCUSSION TITLE
Setting up the "Energy distribution" tile

### DISCUSSION THREAD

**Discussion by bogdanripa** [wlcrs/huawei_solar#340] [Setting up the integration]
Hi. I have everything working, but the visualisation for the "Energy distribution" shows the wrong numbers. This is what I see:

![image](https://user-images.githubusercontent.com/1695752/227775353-ea6917d8-0854-44ac-a43a-9e446346b13d.png)

Where actually I am producing way less and consuming way less as well:

![image](https://user-images.githubusercontent.com/1695752/227775397-74f392dc-b64d-4a73-87eb-d2ca07167fea.png)

My set-up looks as follows:

**configuration.yaml**

```
sensor:
  - platform: template
    sensors:
      grid_consumption:
        friendly_name: "Grid consumption"
        unique_id: sensor.grid_consumption
        unit_of_measurement: "W"
        device_class: power
        value_template: >-
          {% set pmap = states('sensor.power_meter_active_power') | int %}
          {% if pmap < 0 %}
            {{ (- pmap) | int }}
          {% else %}
            {{ (0) | int }}
          {% endif %}
        availability_template: >-
          {{ (states('sensor.power_meter_active_power') not in [ 'unavailable', 'none', 'unknown' ] )}}
      grid_exported:
        friendly_name: "Grid exported"
        unique_id: sensor.grid_exported
        unit_of_measurement: "W"
        device_class: power
        value_template: >-
          {% set pmap = states('sensor.power_meter_active_power') | int %}
          {% if pmap > 0 %}
            {{ (pmap) | int }}
          {% else %}
            {{ (0) | int }}
          {% endif %}
        availability_template: >-
          {{ (states('sensor.power_meter_active_power') not in [ 'unavailable', 'none', 'unknown' ] )}}

utility_meter:
  grid_import_solar_daily_energy:
    source: sensor.grid_consumption
    cycle: daily
  grid_export_solar_daily_energy:
    source: sensor.grid_exported
    cycle: daily
```

**.storage/energy file**

```
{
  "version": 1,
  "minor_version": 1,
  "key": "energy",
  "data": {
    "energy_sources": [
      {
        "type": "grid",
        "flow_from": [
          {
            "stat_energy_from": "sensor.grid_consumption",
            "entity_energy_from": "sensor.grid_consumption",
            "stat_cost": null,
            "entity_energy_price": null,
            "number_energy_price": 0.68
          }
        ],
        "flow_to": [
          {
            "stat_energy_to": "sensor.grid_exported",
            "entity_energy_to": "sensor.grid_exported",
            "stat_compensation": null,
            "entity_energy_price": null,
            "number_energy_price": 0.28
          }
        ],
        "cost_adjustment_day": 0.0
      },
      {
        "type": "solar",
        "stat_energy_from": "sensor.inverter_total_yield",
        "config_entry_solar_forecast": [
          "35a5faee24038b912a3031a3b730a233"
        ]
      },
      {
        "type": "solar",
        "stat_energy_from": "sensor.inverter_daily_yield",
        "config_entry_solar_forecast": null
      }
    ],
    "device_consumption": []
  }
}
```

**Comment by Roving-Ronin** [comment]
Your aware that "sensor.inverter_total_yield" is the total kWh that the inverter has produced in its lifetime (not just for 'today') ?
Not sure if that's what is stuffing you up, as you also have the correct "sensor.inverter_daily_yield" also added into the energy dashboard.

No idea where your getting 117.4kWh production for today from still, given your daily_yield is 41.5kWh, and the lifetime yield is 964.67kWh... so not like they are being summed together. ....?

**Comment by JFHCFC** [comment]
I can't set the entities correctly so that they count correctly, especially the energy storage ...

I tried

    - sensor:
      - name: Bateria ład meas
        state: "{{ [states ('sensor.battery_charge_discharge_power') | round(0), 0] | max}}"

    - sensor:
      - name: Bateria rozład meas
        state: "{{ [states ('sensor.battery_charge_discharge_power') | float*-1 | round(0), 0] | max}}"

    - sensor:
      - name: Bateria ład
        unit_of_measurement: Wh
        state: "{{ states ('sensor.bateria_lad_meas') }}"
        device_class: energy
        state_class: total_increasing


    - sensor:
      - name: Bateria rozład
        unit_of_measurement: Wh
        state: "{{ states ('sensor.bateria_rozlad_meas') }}"
        device_class: energy
        state_class: total_increasing


Bateria ład & Bateria rozład added to energy dashbord but the calculated values are absurd
Where am I going wrong ?

**Reply by Roving-Ronin** [reply]
@JFHCFC sorry I don't have a battery to test against. 
BTW have you read this thread:  https://github.com/wlcrs/huawei_solar/discussions/518
